The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12141392296/job/33853433069) [(plain)](https://github.com/rust-lang/rust/commit/79f9f89c3d93fb914107ef99f55a31c5e7f9a60d/checks/33853433069/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    |                     --- the parameter type `F` must be valid for the anonymous lifetime as defined here...
...
103 |           let r = builder
    |  _________________^
104 | |             .spawn_scoped(s, move || {
105 | |                 rustc_span::create_session_globals_then(edition, Some(sm_inputs), || {
106 | |                     f(CurrentGcx::new())
108 | |             })
108 | |             })
    | |______________^ ...so that the type `F` will meet its required lifetime bounds
    |
    = help: consider adding an explicit lifetime bound `F: '_`...
error[E0311]: the parameter type `R` may not live long enough
   --> compiler/rustc_interface/src/util.rs:103:17
    |
100 |       thread::scope(|s| {
100 |       thread::scope(|s| {
    |                     --- the parameter type `R` must be valid for the anonymous lifetime as defined here...
...
103 |           let r = builder
    |  _________________^
104 | |             .spawn_scoped(s, move || {
105 | |                 rustc_span::create_session_globals_then(edition, Some(sm_inputs), || {
106 | |                     f(CurrentGcx::new())
108 | |             })
108 | |             })
    | |______________^ ...so that the type `R` will meet its required lifetime bounds
    |
    = help: consider adding an explicit lifetime bound `R: '_`...
For more information about this error, try `rustc --explain E0311`.
error: could not compile `rustc_interface` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
Build completed unsuccessfully in 0:10:52
```

</details>

--- a/tests/ui/nll/closure-requirements/thread_scope_correct_implied_bound.rs
+++ b/tests/ui/nll/closure-requirements/thread_scope_correct_implied_bound.rs
@@ -1,0 +1,23 @@
+// This example broke while refactoring the way closure
+// requirements are handled. The setup here matches
+// `thread::scope`.
+
+//@ check-pass
+
+struct Outlives<'hr, 'scope: 'hr>(*mut (&'scope (), &'hr ()));
+impl<'hr, 'scope> Outlives<'hr, 'scope> {
+    fn outlives_hr<T: 'hr>(self) {}
+}
+
+fn takes_closure_implied_bound<'scope>(f: impl for<'hr> FnOnce(Outlives<'hr, 'scope>)) {}
+
+fn requires_external_outlives_hr<T>() {
+    // implied bounds:
+    // - `T: 'scope` as `'scope` is local to this function
+    // - `'scope: 'hr` as it's an implied bound of `Outlives`
+    //
+    // need to prove `T: 'hr` :>
+    takes_closure_implied_bound(|proof| proof.outlives_hr::<T>());
+}
+
+fn main() {}

--- a/tests/ui/nll/closure-requirements/thread_scope_incorrect_implied_bound.rs
+++ b/tests/ui/nll/closure-requirements/thread_scope_incorrect_implied_bound.rs
@@ -1,0 +1,21 @@
+// This example incorrectly compiled while refactoring the way
+// closure requirements are handled.
+
+struct Outlives<'hr: 'scope, 'scope>(*mut (&'scope (), &'hr ()));
+impl<'hr, 'scope> Outlives<'hr, 'scope> {
+    fn outlives_hr<T: 'hr>(self) {}
+}
+
+fn takes_closure_implied_bound<'scope>(f: impl for<'hr> FnOnce(Outlives<'hr, 'scope>)) {}
+
+fn requires_external_outlives_hr<T>() {
+    // implied bounds:
+    // - `T: 'scope` as `'scope` is local to this function
+    // - `'hr: 'scope` as it's an implied bound of `Outlives`
+    //
+    // need to prove `T: 'hr` :<
+    takes_closure_implied_bound(|proof| proof.outlives_hr::<T>());
+    //~^ ERROR the parameter type `T` may not live long enough
+}
+
+fn main() {}

--- a/tests/ui/nll/closure-requirements/thread_scope_incorrect_implied_bound.stderr
+++ b/tests/ui/nll/closure-requirements/thread_scope_incorrect_implied_bound.stderr
@@ -1,0 +1,17 @@
+error[E0310]: the parameter type `T` may not live long enough
+  --> $DIR/thread_scope_incorrect_implied_bound.rs:17:47
+   |
+LL |     takes_closure_implied_bound(|proof| proof.outlives_hr::<T>());
+   |                                               ^^^^^^^^^^^
+   |                                               |
+   |                                               the parameter type `T` must be valid for the static lifetime...
+   |                                               ...so that the type `T` will meet its required lifetime bounds
+   |
+help: consider adding an explicit lifetime bound
+   |
+LL | fn requires_external_outlives_hr<T: 'static>() {
+   |                                   +++++++++
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0310`.

--- a/tests/ui/nll/ty-outlives/projection-implied-bounds.rs
+++ b/tests/ui/nll/ty-outlives/projection-implied-bounds.rs
@@ -1,8 +1,6 @@
 // Test that we can deduce when projections like `T::Item` outlive the
 // function body. Test that this does not imply that `T: 'a` holds.
 
-//@ compile-flags:-Zverbose-internals
-
 use std::cell::Cell;
 
 fn twice<F, T>(mut value: T, mut f: F)

--- a/tests/ui/nll/ty-outlives/projection-implied-bounds.stderr
+++ b/tests/ui/nll/ty-outlives/projection-implied-bounds.stderr
@@ -1,5 +1,5 @@
 error[E0310]: the parameter type `T` may not live long enough
-  --> $DIR/projection-implied-bounds.rs:30:36
+  --> $DIR/projection-implied-bounds.rs:28:36
    |
 LL |     twice(value, |value_ref, item| invoke2(value_ref, item));
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^

